src/makeInfinityAdder.js

```diff
@@ -4,7 +4,22 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
-}
+  let currentResult = 0;
+
+  function adder(number) {
+    if (number === undefined) {
+      const result = currentResult;
+
+      currentResult = 0;
+
+      return result;
+    }
 
+    currentResult += number;
+
+    return adder;
```

Comment:
The `adder` function should be able to chain function calls indefinitely until called with no arguments. To achieve this, ensure that `adder` returns a function when called with an argument. It currently does, but consider adding a comment explaining this behavior for maintainability.